### PR TITLE
perf: replace dynamic regex compilation with byte-level scanning

### LINKED ISSUE #325: perf: cache compiled regexes with OnceLock across validators
## Context
Performance reviewer finding from PR #322 review.

## Problem
Several validators compile regex patterns on every call:
- `find_unique_json_field_line_span` in `hooks/helpers.rs` uses `Regex::new()` per invocation
- `find_unique_matcher_line_span` in `hooks/helpers.rs` compiles regex per call
- Other validators may have similar patterns

## Proposed Solution
Use `std::sync::OnceLock` (or the existing `static_regex!` macro if available) to compile regexes once and reuse them. This is especially impactful for LSP mode where validators run on every keystroke.

## Scope
- Audit all `Regex::new()` calls in `crates/agnix-core/src/rules/`
- Replace with `OnceLock<Regex>` or `static_regex!` pattern
- Benchmark before/after for LSP validation latency

## Labels
enhancement, performance

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

Replaces all 8 dynamic `Regex::new()` calls in auto-fix span-finding helpers with byte-level string scanning functions, eliminating regex compilation overhead entirely.

- New `span_utils` module (`pub(crate)`) with 14 byte-scanning functions
- Replaced dynamic regex in: `hooks/helpers.rs` (4), `rules/mod.rs` (1), `codex.rs` (1), `plugin.rs` (1), `mcp.rs` (1)
- 61 unit tests covering all functions, edge cases (Unicode, CRLF, uniqueness, special chars)
- New benchmark group for auto-fix span finding
- UTF-8 boundary safety with `debug_assert!` checks
- All 2,336 existing tests pass, zero clippy warnings

### Why byte scanning instead of regex caching?

All 8 patterns use runtime-interpolated values via `regex::escape()`, making `static_regex!` / `OnceLock<Regex>` unusable. A `HashMap<String, Regex>` cache was considered but rejected as over-engineering — these patterns search for simple JSON/TOML key-value pairs that are trivially expressible with `str::find()` + whitespace skipping.

## Test Plan

- [x] `cargo test --workspace` — 2,336 tests pass
- [x] `cargo clippy --workspace` — zero warnings
- [x] `cargo build --release` — builds cleanly
- [x] 61 new unit tests in `span_utils` (Unicode, CRLF, edge cases)
- [x] Existing auto-fix integration tests verify identical behavior
- [x] New benchmark: `cargo bench --package agnix-core -- hooks_autofix`

Closes #325

### CODE REVIEW

File: crates/agnix-core/src/span_utils.rs [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+        }
+
+        let val_start = skip_whitespace(bytes, *after_colon);
+        let (_, val_end) = match parse_scalar_at(bytes, val_start) {
+            Some(span) => span,
+            None => continue,
+        };
+
+        let mut end = val_end;
+        end = skip_ws_inline(bytes, end);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/span_utils.rs]: Fixed in 6f3910e - changed to `skip_whitespace` to match original regex `\s*` behavior. Thanks for catching this!

File: crates/agnix-core/src/span_utils.rs [hunk 2]
```diff
@@ -0,0 +1,971 @@
...
+                Some(v) => v,
+                None => continue,
+            };
+
+        if &bytes[inner_start..inner_end] != matcher_value.as_bytes() {
+            continue;
+        }
+
+        let mut end = outer_end;
+        end = skip_ws_inline(bytes, end);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-core/src/span_utils.rs]: Fixed in 6f3910e - changed to `skip_whitespace` to match original regex `\s*` behavior. Thanks!
